hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -309,13 +305,33 @@ private OzoneManager(OzoneConfiguration conf) throws IOException,
       AuthenticationException {
     super(OzoneVersionInfo.OZONE_VERSION_INFO);
     Preconditions.checkNotNull(conf);
-    configuration = conf;
+    configuration = new OzoneConfiguration(conf);
```

Comment:
This function passes a conf object of type OzoneConfiguration. Why are we reallocating a new object before assigining that configuration obect to the member variable?

Reply:
This is being done, so not to change the original values of configuration object passed.

Reply:
But you just lost the reference to the original object. I am slightly confused here.

Reply:
here configuration = new OzoneConfiguration(conf);
And later I am changing the configuration property values in this new object so that it will not affect to original configuration.

I just checked with below:
```

OzoneConfiguration configuration = new OzoneConfiguration();
configuration.set(OZONE_METADATA_DIRS,
        folder.newFolder().getAbsolutePath());

    OzoneConfiguration configuration1 = new OzoneConfiguration(configuration);
    configuration1.set(OZONE_METADATA_DIRS, "bharat");

    System.out.println(configuration.get(OZONE_METADATA_DIRS));
    System.out.println(configuration1.get(OZONE_METADATA_DIRS));
```
/var/folders/g5/fk451xl14vdf891pq7b6m6v00000gp/T/junit8528754098428361111/junit4506171024308775995
bharat

Reply:
Thanks for the catch. This has caused test failure too. Reverted this back.